[Term-Entry]Command-line Bash: Functions (#6310)
Codecademy/docs: content/command-line/concepts/bash/bash.md

# Move a file to a different directory (source path is the first argument, destination path is the second)
mv sourceDirectory/filename.txt destinationDirectory/filename.txt
```

## Bash Functions

> A Bash function is essentially a set of commands that can be called numerous times. The purpose of a _function_ is to help you make 
> your bash scripts more readable, and avoid writing the same code repeatedly.

Reviewer: Avoid using 'you'/'your'. Also, it can be simplified as - 

```suggestion
A **Bash function** is a reusable set of commands that enhances script readability and prevents code duplication.
```

## Example
```shell
  hello() {  
    echo 'Hello, World!'
  }

  # Call the function

  hello 

  # Output:
  # Hello, world!

> ```

Reviewer: Example section should be moved below Syntax section.

## Syntax

Functions may be declared in two different formats:

> 1. The most widely used format is:

Reviewer: ```suggestion
### 1. Standard format
```

```shell 
function_name () {
  commands
}
```

Single line version: 

```shell
function_name () { commands; }
```
 

> 2. The alternative way is using the reversed word ==function==, followed by the function name:

Reviewer: ```suggestion
### 2. Using the `function` keyword
```

```shell
function function_name {
  commands
}
```

Single line version: 

```shell 
function function_name { commands; }
```

> > _Notes when using functions:_ 
> > - Single line commands must end with a semicolon (;) - whether in bash scripts or the terminal directly.
> > - The commands between the curly braces =={ commands; }== are called the functions's _body_. The body can contain any number of `declarations`, `variables`, `loops`, or `conditional statements`.
> > - Use descriptive names for functions.

Reviewer: Follow the syntax as per [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) for adding a note. Enclose code-related terms, keywords, symbols etc in backticks.

```suggestion
> **Note**:
> - Single line commands must end with a semicolon (`;`) - whether in bash scripts or the terminal directly.
> - The commands between the curly braces `{ commands; }` are called the functions's _body_. The body can contain any number of `declarations`, `variables`, `loops`, or `conditional statements`.
> - Use descriptive names for functions.
```

#### How to Declare and Call a Function in the Terminal

To declare and use a function in the terminal:

1. Open the terminal and enter the following:

```shell
my_function () { echo "Hello I'm a function"; echo "Bye!"; }

> ```

Reviewer: This is not needed. It is covered in the syntax, please remove this.

2. Execute (call) the function by entering the function's name in the terminal:

```shell
my_function
```

## Example 

  ```shell
  my_function () {
    echo "Hello I'm a function";
    echo "Bye!"
  }
  
 # Call the function 

  my_function

  # Output:
  # Hello I'm a function
  # Bye!

> ```

Reviewer: Choose and keep only one example in this section.

> The output runs the commands from the function's body

Reviewer: This seems incomplete, please check and correct.
